File: 프로그래머스/lv3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    const visited = [];
    let count = 0;
    
    const dfs = (v) => {
        visited[v] = true;
        
        for(let edge = 0; edge < n; edge++){
            if(!visited[edge] && computers[v][edge]){
                dfs(edge);
            }
        }
    };
    
    for(let v = 0; v < n; v++){
        if(!visited[v]){
            dfs(v);
            count++
        }
    }
    return count
}

File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
def solution(n, computers):
    visited = [False for i in range(n)];
    count = 0;
    
    def dfs(v):
        visited[v] = True;
        
        for edge in range(n):
            if not visited[edge] and computers[v][edge]:
                dfs(edge);
    
    for v in range(n):
        if not visited[v]:
            dfs(v);
            count += 1;
    
    return count

